Xamarin.Forms.Platform.UAP/MediaElementRenderer.cs

```diff
@@ -0,0 +1,236 @@
+	{
+		Windows.System.Display.DisplayRequest _request = new Windows.System.Display.DisplayRequest();
+
+		long _bufferingProgressChangedToken;
+
+		long _positionChangedToken;
+
+		double IMediaElementRenderer.BufferingProgress
+		{
+			get { return Control.BufferingProgress; }
+		}
+
+		TimeSpan? IMediaElementRenderer.NaturalDuration
+		{
+			get
+			{
+				if (Control.NaturalDuration.HasTimeSpan)
+				{
+					return Control.NaturalDuration.TimeSpan;
+				}
+
+				return null;
+			}
+		}
+
+		int IMediaElementRenderer.NaturalVideoHeight
+		{
+			get	{ return Control.NaturalVideoHeight; }
+		}
+
+		int IMediaElementRenderer.NaturalVideoWidth
+		{
+			get { return Control.NaturalVideoWidth; }
+		}
+
+		TimeSpan IMediaElementRenderer.Position
+		{
+			get { return Control.Position; }
+		}
+
+		void IMediaElementRenderer.Seek(TimeSpan time)
+		{
+			Control.Position = time;
+		}
+
+		protected override void OnElementChanged(ElementChangedEventArgs<MediaElement> e)
+		{
+			base.OnElementChanged(e);
+
+			if (e.OldElement != null)
+			{
+				if (Control != null)
+				{
+					if (_positionChangedToken != 0)
+					{
+						Control.UnregisterPropertyChangedCallback(Windows.UI.Xaml.Controls.MediaElement.PositionProperty, _positionChangedToken);
+						_positionChangedToken = 0;
+					}
+
+					Control.CurrentStateChanged -= Control_CurrentStateChanged;
+					Control.SeekCompleted -= Control_SeekCompleted;
+					Control.MediaOpened -= Control_MediaOpened;
+					Control.MediaEnded -= Control_MediaEnded;
+					Control.MediaFailed -= Control_MediaFailed;
+				}
+
+				e.OldElement.SetRenderer(null);
+			}
+
+			if (e.NewElement != null)
+			{
+				SetNativeControl(new Windows.UI.Xaml.Controls.MediaElement());
+				Control.HorizontalAlignment = HorizontalAlignment.Stretch;
+				Control.VerticalAlignment = VerticalAlignment.Stretch;
+
+				e.NewElement.SetRenderer(this);
+				Control.AreTransportControlsEnabled = Element.AreTransportControlsEnabled;
+				Control.AutoPlay = Element.AutoPlay;
+				Control.IsLooping = Element.IsLooping;
+				Control.Stretch = Element.Aspect.ToStretch();
+
+				_bufferingProgressChangedToken = Control.RegisterPropertyChangedCallback(Windows.UI.Xaml.Controls.MediaElement.BufferingProgressProperty, BufferingProgressChanged);
+				_positionChangedToken = Control.RegisterPropertyChangedCallback(Windows.UI.Xaml.Controls.MediaElement.PositionProperty, PositionChanged);
+
+				Control.SeekCompleted += Control_SeekCompleted;
+				Control.CurrentStateChanged += Control_CurrentStateChanged;
+				Control.MediaOpened += Control_MediaOpened;
+				Control.MediaEnded += Control_MediaEnded;
+				Control.MediaFailed += Control_MediaFailed;
+
+				if (Element.Source != null)
+				{
+					Control.Source = Element.Source;
+				}
+			}
+		}
+
+		void Control_MediaFailed(object sender, ExceptionRoutedEventArgs e)
+		{
+			Element?.OnMediaFailed();
+		}
+
+		void Control_MediaEnded(object sender, RoutedEventArgs e)
+		{
+			Element?.OnMediaEnded();
```

Comment:
On Windows you can play from a stopped state at the end of media and it will automatically return to the beginning. The Android code was a trick to enable the same however I've noticed that when you return the video to the beginning it starts buffering the media again which may not be desired so I'm going to change the Android version to add a check when playing to see if the position equals the media duration and if so seek to the beginning and then start playing which with give the same behaviour on all without any preloading and will leave the position at the end of the media as expected.